File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        if(word1.length() != word2.length()) return false;
        int[] freq1 = new int[26];
        int[] freq2 = new int[26];

        HashSet<Character> seen1 = new HashSet<>();
        HashSet<Character> seen2 = new HashSet<>();
        
        for(int i = 0; i<word1.length(); i++) {
            freq1[word1.charAt(i) - 'a']++;
            freq2[word2.charAt(i) - 'a']++;

            seen1.add(word1.charAt(i));
            seen2.add(word2.charAt(i));
        }

        Arrays.sort(freq1);
        Arrays.sort(freq2);

        return Arrays.equals(freq1, freq2) && seen1.equals(seen2);
    }
}

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution(object):
    def closeStrings(self, word1, word2):
        if len(word1) != len(word2): return False
        freq1, freq2 = [0]*26, [0]*26

        seen1 = set()
        seen2 = set()

        for i in range(len(word1)):
            freq1[ord(word1[i]) - ord('a')] += 1
            freq2[ord(word2[i]) - ord('a')] += 1

            seen1.add(word1[i])
            seen2.add(word2[i])

        freq1.sort()
        freq2.sort()

        return freq1 == freq2 and seen1 == seen2

        